sql/testdata/txn

```diff
@@ -92,6 +92,21 @@ SELECT * FROM kv
 ----
 a c
 
+# Abort transaction with a syntax error, and ignore statements until the end of the transaction block
+
+statement ok
+BEGIN
+
+query error syntax error at or near ","
+SELECT COUNT(*, 1) FROM kv
+----
+
+statement error current transaction is aborted, commands ignored until end of transaction block
+UPDATE kv SET v = 'b' WHERE k in ('a')
+
+statement ok
+ROLLBACK
+
```

Comment:
> Shouldn't you also check that the UPDATE did not occur?

Sure, I can add that.

> FYI mysql doesn't abort transactions after a syntax error

Postgres does, though. I'm not attached to either behaviour, but please file an issue if you think this behaviour is worth changing.